### PR TITLE
[serve] Expose deployment statuses in REST API

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
## Why are these changes needed?

This change exposes deployment statuses in Serve's REST API. When the dashboard receives a GET command, it returns both the deployment information and the status information.

## Related issue number

This change builds on the REST API from #22578.

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Unit tests
     - This change adds unit tests to `test_schema.py` and `test_serve_head.py`.

### CODE REVIEW

**Comment by edoakes** [comment]
@shrekris-anyscale you have merge conflicts from the other PR

File: dashboard/modules/serve/serve_head.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    async def get_all_deployments(self, req: Request) -> Response:
+        deployments = list(serve.list_deployments().values())
+        statuses = get_deployment_statuses()
+        serve_instance_schema = serve_instance_to_schema(
+            deployments=deployments, statuses=statuses
+        )
+        return Response(
+            text=json.dumps(serve_instance_schema.json()),
+            content_type="application/json",
+        )
```
Comment (edoakes) [inline dashboard/modules/serve/serve_head.py:37]: this came up in design review; let's separate getting the goal state/config of deployments w/ getting the current status (concretely, split this into a `GET /api/serve/deployments/status` API

this makes it simpler from the caller side: what you `PUT` for the config is exactly what you `GET`

**Review by edoakes — CHANGES_REQUESTED** [review]

**Comment by edoakes** [comment]
@shrekris-anyscale comment about the API, will wait to review the rest until merge conflicts are resolved as it's hard to see the diff from the previous PR.

File: dashboard/modules/serve/serve_head.py (hunk 1 above)
Comment (shrekris-anyscale) [reply dashboard/modules/serve/serve_head.py]: Good call, I updated the API with a separate status endpoint.

File: python/ray/serve/common.py [hunk 2]
```diff
@@ -18,10 +18,10 @@ class EndpointInfo:
...
 
 
-class DeploymentStatus(Enum):
-    UPDATING = 1
-    HEALTHY = 2
-    UNHEALTHY = 3
+class DeploymentStatus(str, Enum):
+    UPDATING = "updating"
+    HEALTHY = "healthy"
+    UNHEALTHY = "unhealthy"
```
Comment (edoakes) [inline python/ray/serve/common.py:24]: nit: use all caps for statuses

**Review by edoakes — APPROVED** [review]

File: python/ray/serve/common.py (hunk 2 above)
Comment (shrekris-anyscale) [reply python/ray/serve/common.py]: Fixed

**Pull request merged**